### ISSUE TITLE
Controling what traits and methods are derived for generated structs.

### ISSUE THREAD

**Issue by TheVova** [danburkert/prost#293]
I could be missing things, but is there a way to customize what derives are added to generated structs?
currently,  the generated structs contain: `#[derive(Clone, PartialEq, ::prost::Message)]`, and the bounds on ` Message: Debug + Send + Sync`, mean the generated struct has to at least be `Clone + Debug + Send + Sync + PartialEq`. `Default` is also generated. 
Firstly are Debug, Send, and Sync actually necessary bounds to implement 'Message'? because doesn't seem to me that they are. 
Secondly, `Default` being generated for each type is actually a double edged sword here. sometime i want to implement locally a custom default for a type, which i cant do because its already generated.
typical use case is i have a proto-message with config options to set. the defaults for protobuf are either 0/None/"" etc, or, if using proto2, what ever is in the [default=xx] annotation. but that sets the default for all messages of this type, which might not be desired. 
I realise Default is needed for protobufs to function, but can we for example, implement something like ProtobufDefault instead? and leave the 'proper' default up to the user?
Lastly, for certain types (basic scalars and enums) prost generates 'getter' methods that just return the value, or default if the value is None... This is in my opinion a mistake. It generates an inconsistent API (i.e why can i do self.field() if field is i64, but i cant do self.field() if field is Option<NestedStruct>?)
and interferes with generating custom accessor implementations by the user.

In short, would the ability to disable anything other than the bare minimum for dealing with protobufs be desirable? If so, im going to give it a go :)

**Comment by danburkert** [comment]
The set of derives can be added to, but there is no way to remove derives.

As far as all of your other concerns, none of those things are subject to change.  I recommend using wrapper types if you have significant logic that you want to layer on to `prost` generated types.
